Больше подробностей (часть 1)

RuslanPro01 opened the pull request:
---
:mortar_board: [Больше подробностей (часть 1)](https://up.htmlacademy.ru/react/12/user/854371/tasks/8)

Commit:
feat: Реализует показ видеоплеера.

- научит карточку фильма в зависимости от стейта показывать видеоплеер

komaritskaya commented on project/src/common-const.ts:
```diff
@@ -14,3 +14,5 @@ export const Path = {
   PageNotFound: '*'
 } as const;
+
+export const timeOutActiveVideo = 1000;
```
Похоже на константу (см критерий [Б7](https://up.htmlacademy.ru/react/12/criteries#b7))

komaritskaya commented on project/src/components/film-cards/film-cards.tsx:
```diff
@@ -1,27 +1,46 @@
 
 function FilmCards({films}: {films: Films}): JSX.Element {
   const [activeFilm, setActiveFilm] = useState<null | number>(null);
+  const [isActiveFilm, setIsActiveFilm] = useState<boolean>(false);
```
Тут два момента:
Во-первых, не самое удачное именование: `activeFilm` и `isActiveFilm` - очень легко запутаться.
Во-вторых, как и в прошлом задании, лучше по-другому распределить ответственность между компонентами. `FilmCards` должен только устанавливать айдишник активного фильма, как мы и делали до этого. Все остальное (плееры, таймауты и т.д.) - не его зона ответственности. Здесь мы должны внутри `FilmCard` подписываться на изменение пропса `isActiveFilm` и запускать либо останавливать плеер. Т.е. `isNeedVideoPlayer` должен стать внутренним состоянием `film-card`. Я бы, конечно, еще упростила название и поменяла бы его на `isPlaying`, но это уже мелочи)

Commit:
feat: Добавит библиотеки Helmet и classNames.

- Установит библиотеки
- Добавит на каждую страницу `title`.
- начнет использование classNames в film-card.tsx

Commit:
refactor: Разделит ответственность для показа видео
